hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSourceWALReader.java

```diff
@@ -166,6 +169,9 @@ public void run() {
       } catch (InterruptedException e) {
         LOG.trace("Interrupted while sleeping between WAL reads or adding WAL batch to ship queue");
         Thread.currentThread().interrupt();
+      } finally {
```

Comment:
What is the advantage here to not use try-with-resources but a try finally?

Reply:
I want to use `entryStream` object in handleEofException method. If I use try-wth-resources, I don't have access to `entryStream`

Reply:
OK, so if we could move the main logic in WALEntryStream.getArchivedLog to a util class, then we do not need try finally then? In general, I do not think we should expose WALEntryStream.getArchivedLog directly. Let's add a public static method in AbstractFSWALProvider, just below the getWALArchiveDirectoryName method. I think this is the right place for holding a public method.

Reply:
Thank you @Apache9  for the review. Didn't know a utility method [AbstractFSWALProvider#getArchivedLogPath ](https://github.com/apache/hbase/blob/master/hbase-server/src/main/java/org/apache/hadoop/hbase/wal/AbstractFSWALProvider.java#L480) already exists. 
Removed all the changes to make WALEntryStream class. Please review again.